pkg/acsengine/defaults.go

```diff
@@ -675,6 +668,16 @@ func setStorageDefaults(a *api.Properties) {
 	}
 }
 
+func setHostedMasterOverrides(a *api.Properties) {
+	a.HostedMasterProfile.Subnet = DefaultKubernetesMasterSubnet
+
+	for _, profile := range a.AgentPoolProfiles {
+		if !a.OrchestratorProfile.IsOpenShift() {
+			profile.Distro = api.Ubuntu
```

Comment:
I think we might want to do this in `setAgentNetworkDefaults` instead like so: 
```
		// don't default Distro for OpenShift
		if !a.OrchestratorProfile.IsOpenShift() {
			if profile.Distro == "" {
                                    if cs.Properties.HostedMasterProfile != nil { 
                                                profile.Distro = api.Ubuntu
                                    } else {
				                profile.Distro = api.AKS
                                    }
			}
		}
```

Reply:
I'm worried about overriding the value no matter what because this might override the value for a scale or an upgrade if we do enable the AKS distro progressively

Reply:
Also it's more readable to have both default values in one place IMO